include/simsimd/probability.h

```diff
@@ -219,12 +221,14 @@ SIMSIMD_PUBLIC void simsimd_js_f32_neon(simsimd_f32_t const *a, simsimd_f32_t co
     float32x4_t log_ratio_b_vec = _simsimd_log2_f32_neon(ratio_b_vec);
     float32x4_t prod_a_vec = vmulq_f32(a_vec, log_ratio_a_vec);
     float32x4_t prod_b_vec = vmulq_f32(b_vec, log_ratio_b_vec);
+    
     sum_vec = vaddq_f32(sum_vec, vaddq_f32(prod_a_vec, prod_b_vec));
     if (n != 0) goto simsimd_js_f32_neon_cycle;
 
     simsimd_f32_t log2_normalizer = 0.693147181f;
+    sum_vec = vmulq_f32(sum_vec, half_vec);
     simsimd_f32_t sum = vaddvq_f32(sum_vec) * log2_normalizer;
-    *result = sum / 2;
+    *result = SIMSIMD_SQRT(sum);
 }
 
```

Comment:
Check out the vmulq_n_f32 intrinsic on Arm 🤗

Reply:
The mul intrinsic is for making the sqrt faster ? I shall take a look also. Thanks !

Reply:
No, I just pressed the wrong line in the GUI, sorry for that. Look into `vmulq_n_f32` to replace some of your use of the `vmulq_f32`. It fits better 🤗

Reply:
I tried to figure out if multiplying only `sum_vec` at the end by 0.5 using `vmulq_n_f32` is better, but multiplying `prod_a_vec` and `prod_b_vec` by 0.5 before summing them up seems to be a bit more precise. 

I get upto 3 digits of precision (default in the python test seems to be 7). Should I adjust the precision while testing to 1e-03 or investigate how to get more precision ?

Reply:
The error shouldn't be coming from multiplication. It's likely rooted in the quality of the logarithm implementation. I'm currently working on improved variants as part of #220 🤗